playground-core/core-assertions.js

```diff
@@ -0,0 +1,661 @@
+/* *******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the W3C Software Notice and
+ * Document License (2015-05-13) which is available at
+ * https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR W3C-20150513
+ ********************************************************************************/
+
+ /**
+  * This file contains functions, which are required by the core package as
+  * well as by the playground-assertions package
+  */
+
+// A special JSON validator that is used only to check whether the given object has duplicate keys.
+// The standard library doesn't detect duplicate keys and overwrites the first one with the second one.
+// TODO: replace with jsonlint ??
+const jsonValidator = require('json-dup-key-validator')
+
+// This is used to validate if the multi language JSON keys are valid according to the BCP47 spec
+const bcp47pattern = /^(?:(en-GB-oed|i-ami|i-bnn|i-default|i-enochian|i-hak|i-klingon|i-lux|i-mingo|i-navajo|i-pwn|i-tao|i-tay|i-tsu|sgn-BE-FR|sgn-BE-NL|sgn-CH-DE)|(art-lojban|cel-gaulish|no-bok|no-nyn|zh-guoyu|zh-hakka|zh-min|zh-min-nan|zh-xiang))$|^((?:[a-z]{2,3}(?:(?:-[a-z]{3}){1,3})?)|[a-z]{4}|[a-z]{5,8})(?:-([a-z]{4}))?(?:-([a-z]{2}|\d{3}))?((?:-(?:[\da-z]{5,8}|\d[\da-z]{3}))*)?((?:-[\da-wy-z](?:-[\da-z]{2,8})+)*)?(-x(?:-[\da-z]{1,8})+)?$|^(x(?:-[\da-z]{1,8})+)$/i // eslint-disable-line max-len
+
+
+module.exports =  {
+    checkPropUniqueness,
+    checkSecurity,
+    checkMultiLangConsistency
+}
+
+
+// -------------------------------------------------- checkPropUniqueness
+
+/**
+ *  Checking whether in one interaction pattern there are duplicate names, e.g. two properties called temp
+ *  However, if there are no properties then it is not-impl
+ *
+ * @param {string} tdString The Td under test as string
+ */
+function checkPropUniqueness(tdString) {
+
+    const results = []
+
+    // jsonvalidator throws an error if there are duplicate names in the interaction level
+    try {
+        jsonValidator.parse(tdString, false)
+
+        if (noAssertions) {
```

Comment:
`noAssertions` is not defined?

Reply:
Removed the conditional statement (wasn't required anymore, but did not change anything since `noAssertions` was never true).